### PR TITLE
refine logging in pkg/webhook/inject

### LINKED ISSUE #2891: Refine logging in `pkg/webhook/inject`, refactor these function as methods of a Struct, them move package level variable `log` as a field

### PR DESCRIPTION
**Pull request by STRRL**
Signed-off-by: STRRL <str_ruiling@outlook.com>

### What problem does this PR solve?

- close #2891

### What's changed and how it works?

- replace package level logger with `log.L()`

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [x] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [x] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

**Review by afzal442 — APPROVED** [review]
/LGTM 😄

File: pkg/webhook/inject/inject.go [hunk 1]
```diff
@@ -52,7 +50,7 @@ const (
 func Inject(res *v1.AdmissionRequest, cli client.Client, cfg *config.Config, controllerCfg *controllerCfg.ChaosControllerConfig, metrics *metrics.ChaosControllerManagerMetricsCollector) *v1.AdmissionResponse {
 	var pod corev1.Pod
 	if err := json.Unmarshal(res.Object.Raw, &pod); err != nil {
-		log.Error(err, "Could not unmarshal raw object")
+		log.L().WithName("inject").Error(err, "Could not unmarshal raw object")
```
Comment (WangXiangUSTC) [inline pkg/webhook/inject/inject.go:53]: how about defining a variable for `log.L().WithName("inject")`?

Comment (STRRL) [reply pkg/webhook/inject/inject.go]: Updated! PTAl

**Comment by STRRL** [comment]
duplicated with https://github.com/chaos-mesh/chaos-mesh/pull/2917, I am going to close this PR

**Pull request closed without merging**